lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -306,15 +310,16 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
   defp unavailable_secret?(_socket, ""), do: false
 
   defp unavailable_secret?(socket, preselect_name) do
-    not session?(socket, preselect_name) and
-      not app?(socket, preselect_name) and
-      not hub?(socket, preselect_name)
+    get_secret(socket, preselect_name) != []
   end
 
   defp title(%{assigns: %{select_secret_ref: nil}}), do: "Add secret"
   defp title(%{assigns: %{select_secret_options: %{"title" => title}}}), do: title
   defp title(_), do: "Select secret"
 
+  defp build_origin(%{"secret_name" => _, "secret_origin" => "session"}), do: :session
+  defp build_origin(%{"secret_name" => _, "secret_origin" => "app"}), do: :app
+  defp build_origin(%{"secret_name" => _, "secret_origin" => id}), do: {:hub, id}
```

Comment:
The idea is to have a single definition of `build_origin` that works for both scenarios. So you want to normalize all callers of `build_origin` to always pass parameters in the same shape (i.e. either "store"+"hub_id" OR "secret_origin").